secret/get_secret.go

```diff
@@ -71,7 +83,7 @@ func GetSecretByRefOrLabel(ctx context.Context, clt client.Client, ref *corev1.O
 	}
 
 	secretList := &corev1.SecretList{}
-	if err = clt.List(ctx, secretList, client.InNamespace(objKey.Namespace), client.MatchingLabelsSelector{Selector: selector}); err != nil {
+	if err = clt.List(ctx, secretList, client.InNamespace(ns), client.MatchingLabelsSelector{Selector: selector}); err != nil {
```

Comment:
主要修改是这个查询的 ns，并不完全是 secret 的 namespace。
因为存在一些夸命名空间同步的凭据。

Reply:
那为什么不是调用方，直接把这个ns 传入到 ref *corev1.ObjectReference 里边呢？ 感觉上让  GetSecretByRefOrLabel 去包含  If the ref.namespace is empty, it will use the namespace from the ctx. 的逻辑，稍微有点牵强。

Reply:
因为我们支持用户只配置 secret 的 name，默认在资源同名 ns 下选取该 secret。
所以会有这么个逻辑。